torch/ao/quantization/fx/README.md

```diff
@@ -0,0 +1,407 @@
+                      Calibrate/Train
+                              |
+(convert_fx)                  |
+—--------------------------------------------------------
+|                         Convert                       |
+|                        Lowering                       |
+—--------------------------------------------------------
+                              |
+                       Quantized Model
+
+Please refer to TODO: link [(draft) README.md of quantization syntax transforms] for definitions of terminologies.
+
+## Overview
+The FX graph representation is pretty close to python/eager mode, it preserves many python/eager mode constructs like modules, functionals, torch ops, so overall the implementation reuses some of building blocks and utilities from eager mode quantization, this includes the QConfig, QConfig propagation (might be removed), fused modules, QAT module, quantized modules, QAT module swapping utility. Also the overall flow exactly matches eager mode quantization, the only difference is that the transformations like fusion, inserting stubs are fully automated and controlled by QConfigMapping and BackendConfig.
+
+### Quantization Flow Code Example
+```
+m = M().eval()
+qconfig = ...
+qconfig_mapping = get_default_qconfig_mapping(“fbgemm”)
+backend_config = get_default_backend_config(“fbgemm”)
+m = prepare_fx(m, qconfig_mapping, backend_config=backend_config)
+# or m = prepare_qat_fx(m, qconfig_mapping, backend_config=backend_config)
+# calibration/training
+...
+m = convert_fx(m, backend_config=backend_config)
+```
+
+## High Level Flow with Simple Example
+
+Floating Point Model — (1.1 `_fuse_fx`) —> Fused Model —-- (1.2 QAT Module Swap) —--> Model with QAT modules —-- (1.3 Insert Observers) —---> Prepared Model
+---- (2.1 `convert_to_reference`) —----> Reference Quantized Model —---- (2.2 Lower to Native Backend) —-----> Quantized Model
+
+`prepare_fx`: Step 1.1 to 1.3
+`convert_fx`: Step 2.1 to 2.2
+
+In the following, I’ll first have a detailed description for each step, and then talk about the corresponding settings in BackendConfig. We’ll follow the terminologies defined in (draft) README.md of quantization syntax transforms in this doc. 
+
+### 0. Original Model
+
+```
+class LinearReLUModule(torch.nn.Module):
+   def __init__(self):
+       super().__init__()
+       self.linear = torch.nn.Linear(5, 10).float()
+       self.relu = torch.nn.ReLU()
+ 
+   def forward(self, x):
+       return self.relu(self.linear(x))
+```
+
+### 1.1 Fusion
+```
+fused: GraphModule(
+  (linear): LinearReLU(
+    (0): Linear(in_features=5, out_features=10, bias=True)
+    (1): ReLU()
+  )
+)
+
+def forward(self, x):
+    linear = self.linear(x);  x = None
+    return linear
+```
+
+In eager mode quantization, users need to call `fuse_modules` or `fuse_modules_qat` and provide a list of fully qualified names for the submodules that need to be fused. We will get the same result as the eager mode fusion.
```

Comment:
Would it make sense to remove the comparisons with eager mode from this doc, and just explain what's happening?  This way people who don't understand eager mode would be able to follow better.